### PR TITLE
Odyssey Stats: Add a Jetpack upsell section to the traffic page

### LINKED ISSUE #71820: Odyssey Stats: Restore Jetpack upsell section
Image

Restore the Jetpack upsell section that used to exist in the old Jetpack Stats.

### PR DESCRIPTION
**Pull request by jsnmoon**
Fixes #71820.

Design|Implementation
-|-
Image|image
image|image

## Proposed Changes

Integrates a new Jetpack Upsell Card into a new section on the Odyssey Stats Traffic page. Available upsells are determined by excluding currently active products for the Jetpack site; the corresponding CTA buttons link to the WordPress.com checkout pages.

## Testing Instructions

* Apply this change to your Odyssey Stats instance using option 1 or 2 from PejTkB-3E-p2.
* Open Odyssey Stats and navigate to the bottom of the traffic tab.
* Ensure you see the new Jetpack upsell section.
* Spin up the component package storybook and ensure the Jetpack Upsell Card stories work as expected.
(`yarn workspace @automattic/components run storybook`)

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/stats/jetpack-upsell-section/use-purchased-products.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	[ 'social', JETPACK_SOCIAL_PRODUCTS as readonly string[] ],
+	[ 'video', JETPACK_VIDEOPRESS_PRODUCTS as readonly string[] ],
+] );
+
+function formatResponse( responseData?: Record< string, string >[] ) {
+	const products = [] as string[];
+	// Find active purchase product slugs.
+	const purchasedProductSlugs =
+		responseData?.filter( ( p ) => p.active === '1' )?.map( ( p ) => p.product_slug ) ?? [];
+	// Append active product sulgs to the products array.
```
Comment (dognose24) [inline client/my-sites/stats/jetpack-upsell-section/use-purchased-products.tsx]:
```suggestion
	// Append active product slugs to the products array.
```

**Review by dognose24 — APPROVED** [review]
Works as described! @jsnmoon Great work! The only concern is the mobile layout for the upsell card, which may be a minor scenario for Odyssey Stats.

截圖 2023-03-02 上午1 28 57

File: client/my-sites/stats/jetpack-upsell-section/use-purchased-products.tsx (hunk 1 above)
Comment (jsnmoon) [reply client/my-sites/stats/jetpack-upsell-section/use-purchased-products.tsx]: Good catch!

**Comment by jsnmoon** [comment]
> Works as described! @jsnmoon Great work! The only concern is the mobile layout for the upsell card, which may be a minor scenario for Odyssey Stats.

Thanks for the heads up, @dognose24. I went ahead and implemented the mobile layout according to the design specs in Figma:

Design|Implementation
-|-
image|image

**Review by dognose24 — APPROVED** [review]
LGTM! 👍🏼

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7839416

Thank you @jsnmoon for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7839416).

**Pull request merged**